File: docs/src/examples/simple.jl
using Korg
using PyPlot
using FormationTemps; FT = FormationTemps

# get the linelist
linelist = Korg.read_linelist(joinpath(FT.datdir, "Sun_VALD.lin"))[16000:16100]

# set stellar parameters
Teff = 5777.0
logg = 4.44
A_X = Korg.asplund_2020_solar_abundances
Fe_H = 0.0
vsini = 2100.0
ζ_RT = 3400.0   # radial-tangential macroturbulent broadening 
ξ = 850.0       # microturbulent broadenign

# create StellarProps composite type to hold everything 
star_props = StellarProps(Teff=Teff, logg=logg, Fe_H=Fe_H, 
                          vsini=vsini, v_macro=ζ_RT, v_micro=ξ)

# get the flux + formation temperature spectra
form_temp_result = FT.calc_formation_temp(star_props, linelist; Δλ=0.01)

# parse the result
wavs = form_temp_result.wavs
flux = form_temp_result.flux
temp = form_temp_result.form_temps

# plot the result
fig, ax1 = plt.subplots(figsize=(9.6,4.8))
ax1.plot(wavs, temp, c="k")
ax1.set_xlabel("Vacuum Wavelength [Å]")
ax1.set_ylabel("Formation Temperature [K]")
fname = joinpath(FT.moddir, "docs", "src", "static", "temp_example_jl.png")
fig.savefig(fname, bbox_inches="tight")
plt.show()
plt.clf(); plt.close()

File: docs/src/examples/simple.py
import matplotlib
matplotlib.use("Agg")

from juliacall import Main as jl
import numpy as np
import matplotlib.pyplot as plt

# load FormationTemps
jl.seval("using Korg")
jl.seval("using FormationTemps")
FT = jl.FormationTemps
Korg = jl.Korg

# read the linelist (ships with the package)
# slicing done on Julia side to preserve 1-based indexing
linelist = jl.seval(
    'Korg.read_linelist(joinpath(FormationTemps.datdir, "Sun_VALD.lin"))[16000:16100]'
)

# set stellar parameters (velocities in m/s)
star = FT.StellarProps(Teff=5777.0, logg=4.44, Fe_H=0.0,
                       vsini=2100.0, v_macro=3400.0, v_micro=850.0)

# compute formation temperatures (use_gpu=False for portability)
result = FT.calc_formation_temp(star, linelist, use_gpu=False, convolve=True, u1=0.43, u2=0.31)

# extract results into numpy arrays
wavs = np.asarray(result.wavs)
flux = np.asarray(result.flux)
temps = np.asarray(result.form_temps)

# plot
fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(9.6, 6.4), sharex=True)
ax1.plot(wavs, flux, c="k")
ax1.set_ylabel("Normalized Flux")
ax2.plot(wavs, temps, c="k")
ax2.set_xlabel("Vacuum Wavelength [Ang]")
ax2.set_ylabel("Formation Temperature [K]")
fig.tight_layout()
fig.savefig("formation_temps.png", dpi=150, bbox_inches="tight")
print("Saved formation_temps.png")
